webapp/src/js/services/contact-view-model-generator.js

```diff
@@ -264,28 +264,19 @@ angular.module('inboxServices').factory('ContactViewModelGenerator',
         });
     };
 
-    return function(id, options) {
-      return LineageModelGenerator.contact(id, options)
-        .then(function(model) {
-          setPrimaryContact(model);
-          setSchemaFields(model);
-          setMutedState(model);
+    return {
+      getContact: function(id, options) {
+        return LineageModelGenerator.contact(id, options)
+          .then(function(model) {
+            setPrimaryContact(model);
+            setSchemaFields(model);
+            setMutedState(model);
 
-          model.loadingChildren = true;
-          model.loadingReports = true;
-          model.reportLoader = loadChildren(model, options)
-            .then(children => {
-              model.children = children;
-              model.loadingChildren = false;
-              return loadReports(model);
-            })
-            .then(reports => {
-              model.reports = reports;
-              model.loadingReports = false;
-            });
-
-          return model;
-        });
```

Comment:
This had to be updated as `reportLoader` was mutating `model` (by which point it was in the store as `selected`), and in redux everything has to be immutable otherwise things break. As far as I can tell the new api makes sense, although the `options` passing here is not ideal, not sure what a better approach would be.